go/ql/src/experimental/frameworks/DecompressionBombsCustomizations.qll

```diff
@@ -0,0 +1,772 @@
+   */
+  module KlauspostZlib {
+    class TheSink extends Sink {
+      TheSink() {
+        exists(Method m, DataFlow::CallNode cn |
+          m.hasQualifiedName("github.com/klauspost/compress/zlib", "reader", "Read") and
+          cn = m.getACall()
+        |
+          this = cn.getReceiver() and
+          not hasFlowToComparison(cn.getResult(0))
+        )
+      }
+    }
+
+    class TheAdditionalTaintStep extends AdditionalTaintStep {
+      TheAdditionalTaintStep() { this = "AdditionalTaintStep" }
+
+      override predicate isAdditionalFlowStep(
+        DataFlow::Node fromNode, FlowState fromState, DataFlow::Node toNode, FlowState toState
+      ) {
+        exists(Function f, DataFlow::CallNode call |
+          f.hasQualifiedName("github.com/klauspost/compress/zlib", "NewReader") and
+          call = f.getACall()
+        |
+          fromNode = call.getArgument(0) and
+          toNode = call.getResult(0) and
+          fromState = "" and
+          toState = "ZlibNewReader"
+        )
+      }
+
+      override predicate isAdditionalFlowStep(DataFlow::Node fromNode, DataFlow::Node toNode) {
+        none()
+      }
+    }
+  }
+
+  /**
+   * Provides decompression bomb sinks and additional flow steps for `compress/zlib` package
+   */
+  module CompressZlibBombs {
+    class TheSink extends Sink {
+      TheSink() {
+        exists(Method m, DataFlow::CallNode cn |
+          m.hasQualifiedName("compress/zlib", "reader", "Read") and
+          cn = m.getACall()
+        |
+          this = cn.getReceiver() and
+          not hasFlowToComparison(cn.getResult(0))
+        )
+      }
+    }
+
+    class TheAdditionalTaintStep extends AdditionalTaintStep {
+      TheAdditionalTaintStep() { this = "AdditionalTaintStep" }
+
+      override predicate isAdditionalFlowStep(
+        DataFlow::Node fromNode, FlowState fromState, DataFlow::Node toNode, FlowState toState
+      ) {
+        exists(Function f, DataFlow::CallNode call |
+          f.hasQualifiedName("compress/zlib", "NewReader") and
+          call = f.getACall()
+        |
+          fromNode = call.getArgument(0) and
+          toNode = call.getResult(0) and
+          fromState = "" and
+          toState = "ZlibNewReader"
+        )
+      }
+
+      override predicate isAdditionalFlowStep(DataFlow::Node fromNode, DataFlow::Node toNode) {
+        none()
+      }
+    }
+  }
+
+  /**
+   * Provides decompression bomb sinks and additional flow steps for `github.com/golang/snappy` package
+   */
+  module GolangSnappy {
+    class TheSink extends Sink {
+      TheSink() {
+        exists(Method m, DataFlow::CallNode cn |
+          m.hasQualifiedName("github.com/golang/snappy", "Reader", ["Read", "ReadByte"]) and
+          cn = m.getACall()
+        |
+          this = cn.getReceiver() and
+          not hasFlowToComparison(cn.getResult(0))
+        )
+      }
```

Comment:
The [docs for snappy](https://pkg.go.dev/github.com/golang/snappy#Reader.Read) say that these two methods just implement interface methods from the standard library, one of which you model. I think it would be better to just model those interface methods. Please check if the same is true for the other libraries. (Annoyingly it isn't very clear in Go if a method is implementing an interface method, but there are limited interface methods that make sense here so you can just compare with those function names and signatures.)